### PR TITLE
Implement Record Specification

### PR DESCRIPTION
**Pull request by dwedul-figure**
## Description

Implement the Record Specification stuff.

closes: #59

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: proto/provenance/metadata/v0/types.proto [hunk 1]
```diff
@@ -98,15 +98,15 @@ enum PartyType {
...
+  string       updated_by   = 4;
+  int32        version      = 5;
+  string       message      = 6;
 }
 
 // proto 3 timestamp structure (hack to get around gogo issues with invalid dates)
 message GogoTimeHack {
   int64 seconds = 1;
-  int32 nanos = 2;
+  int32 nanos   = 2;
```
Comment (dwedul-figure) [inline proto/provenance/metadata/v0/types.proto:111]: These are the result of a `make proto-format` command.

Comment (iramiller) [reply proto/provenance/metadata/v0/types.proto:111]: Yeah... that shouldn’t have merged in that way... I think our proto actions might be a little off cause they should have flagged that.

**Review by channa-figure — COMMENTED** [review]
I think we are missing message handling needed in the handler.go file?

**Comment by dwedul-figure** [comment]
> I think we are missing message handling needed in the handler.go file?

Yup. Fixed now.

**Review by channa-figure — APPROVED** [review]

**Pull request merged**